src/main/java/com/microsoft/lst_bench/client/QueryResult.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright (c) Microsoft Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.microsoft.lst_bench.client;
+
+import java.sql.ResultSet;
+import java.sql.ResultSetMetaData;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Represents the query result of a query issued against a source. Query result entries should be
+ * mapped to column name -> list of column values.
+ */
+public class QueryResult {
+
+  private final Map<String, List<Object>> valueList;
+
+  public QueryResult() {
+    this.valueList = new HashMap<>();
+  }
+
+  public void populate(ResultSet rs) throws SQLException {
```

Comment:
This method still populates the internal structure eagerly. Please leave a TODO and create a GH issue (it seems a good enhancement that could be taken care of by any new contributor).

Reply:
Added the todo, will create the GH issue once this is pushed.